arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/option.kt

```diff
@@ -0,0 +1,30 @@
+package arrow.core.continuations
+
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.identity
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+import kotlin.jvm.JvmInline
+
+public suspend inline fun <A> option(crossinline f: suspend OptionEffectScope.() -> A): Option<A> =
```

Comment:
I think we need`option { }` and `option.eager { }` to replace the existing behavior in `arrow.core.computations`.

Reply:
I've exposed an eager version, but that would only point to eagerEffect with no dsl changes on top as in `option { }`
We could do it if the eagerEffect `shift` is not suspend.

Reply:
Updated it with an dsl